src/simulation_strategies/FixedLossSimulationStrategy.py

```diff
@@ -31,10 +33,13 @@ def __prepare_initial_state(self, input_state: ndarray) -> ndarray:
         """
             This method is used to prepare psi_0 state (formula 23 from ref. [1]).
             :param input_state: Initial lossy bosonic state.
-            :return:
+            :return: Returns initial state of the formula (being an equal superposition
+            on n photons 'smeared' on first n modes).
         """
-        initial_number_of_photons = sum(input_state)
-        prepared_state = input_state[:]
+        initial_number_of_photons = int(sum(input_state))
+        prepared_state = [1 for _ in range(initial_number_of_photons)]
```

Comment:
Alternative expression:
```
prepared_state = [1] * initial_number_of_photons
prepared_state.extend([0] * (self.number_of_observed_modes - initial_number_of_photons))
```

Reply:
Is it anyhow better / faster / more common? If so I'll change it.

Reply:
Just an alternative. I find it easier to read, but it doesn't really matter.